tasks(sep/base): add Base Sepolia MCP L1 playbook (#229)
ethereum-optimism/superchain-ops: tasks/sep/base-001-MCP-L1/VALIDATION.md

# Validation

This document can be used to validate the state diff resulting from the execution of the upgrade
transaction.

For each contract listed in the state diff, please verify that no contracts or state changes shown in the Tenderly diff are missing from this document. Additionally, please verify that for each contract:

- The following state changes (and none others) are made to that contract. This validates that no unexpected state changes occur.
- All addresses (in section headers and storage values) match the provided name, using the Etherscan and Superchain Registry links provided. This validates the bytecode deployed at the addresses contains the correct logic.
- All key values match the semantic meaning provided, which can be validated using the storage layout links provided.

## State Overrides

The following state overrides should be seen:

> ### `0x0fe884546476dDd290eC46318785046ef68a0BA9` (The 1 of 1 `ProxyAdmin` owner Safe)

Reviewer: Looks like the safe Base uses here is a 3/7. This comment is probably residual from conduit's which was a 1/1 at the time. I would just relace "The 1 of 1 `ProxyAdmin` owner Safe" with "The `ProxyAdmin` owner Safe" (occurs here and below)

- **Key:** `0x0000000000000000000000000000000000000000000000000000000000000004` <br/>
  **Before:** `0x000000000000000000000000da3037ff70ac92cd867c683bd807e5a484857405` <br/>
  **After:** `0x0000000000000000000000000000000000000000000000000000000000000078` <br/>
  **Meaning:** Sets `submissionInterval` at slot `0x04` (4). The correctness of
   this slot is attested to in the Optimism repo at [storageLayout/L2OutputOracle.json](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.3.0/packages/contracts-bedrock/snapshots/storageLayout/L2OutputOracle.json#L38-L42). `0x78` is 120 in decimal, which matches the current value found by `cast call 0x84457ca9d0163fbc4bbfe4dfbb20ba46e48df254 "SUBMISSION_INTERVAL()(uint256)" -r https://ethereum-sepolia.publicnode.com`.

- **Key:** `0x0000000000000000000000000000000000000000000000000000000000000005` <br/>

>   **Before:** `0x00000000000000000000000020044a0d104e9e788a0c984a2b7eae615afd046b` <br/>

Reviewer: In other MCP upgrade, I believe these both of these `before` keys are zero? If so, let's take a closer look to understand what's being overwritten here and see if it's safe

Reviewer: Based on the below slot changes, it looks like the value in slot 4 was the challenger address, and the value in slot 5 was proposer address. These are now in slots 6 and 7 respectively.

This change is surprising to me, since it implies we had a storage layout change?

Author: Here is Base Sepolia's [initially deployed commit](https://github.com/base-org/contract-deployments/blob/26783846e57316b02ed511d75edf4bf92dcc4269/sepolia/2023-09-26-deploy/.env#L1C11-L1C51) vs the [MCP contract release commit](https://github.com/ethereum-optimism/optimism/commit/e6ef3a900c42c8722e72c2e2314027f85d12ced5) [compared](https://github.com/ethereum-optimism/optimism/compare/a7ff5a811612fa338d0a6d6dd72dc2ec9badef6d..e6ef3a900c42c8722e72c2e2314027f85d12ced5#diff-f92c9308509097bf11b443e68883b1ed5a542f9d306988cdec0431831d9f1f5a).

If you look at the diff of the `L2OutputOracle.sol` and `L2OutputOracle.json`, it does look like the storage slots moved around and some new ones were added.

Author: Confirmed that there are storage layout changes here, but all moved variables are reset in `initialize`, so this change is okay.

   this slot is attested to in the Optimism repo at [storageLayout/L2OutputOracle.json](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.3.0/packages/contracts-bedrock/snapshots/storageLayout/L2OutputOracle.json#L52-L56). This value matches the current address found by `cast call 0x84457ca9d0163fbc4bbfe4dfbb20ba46e48df254 "CHALLENGER()(address)" -r https://ethereum-sepolia.publicnode.com`.

- **Key:** `0x0000000000000000000000000000000000000000000000000000000000000007` <br/>
  **Before:** `0x0000000000000000000000000000000000000000000000000000000000000000` <br/>
  **After:** `0x00000000000000000000000020044a0d104e9e788a0c984a2b7eae615afd046b` <br/>
  **Meaning:** Sets `proposer` at slot `0x07` (7). The correctness of
   this slot is attested to in the Optimism repo at [storageLayout/L2OutputOracle.json](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.3.0/packages/contracts-bedrock/snapshots/storageLayout/L2OutputOracle.json#L59-L63). This value matches the current address found by `cast call 0x84457ca9d0163fbc4bbfe4dfbb20ba46e48df254 "PROPOSER()(address)" -r https://ethereum-sepolia.publicnode.com`.

> todo: this one differs from the template as well, the finalizationPeriodSeconds is set to 12 (0xc) on Base Sepolia, the template said: Units are in seconds, so the value should be `0x93a80` which is 604,800 in decimal.

Reviewer: it looks like this is in fact the Base current value, so this difference seems ok

  **Meaning:** Sets `finalizationPeriodSeconds` at slot `0x08` (8). The correctness of
   this slot is attested to in the Optimism repo at [storageLayout/L2OutputOracle.json](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.3.0/packages/contracts-bedrock/snapshots/storageLayout/L2OutputOracle.json#L66-L70). Units are in seconds, so the value should be `0xc` which is 12 in decimal. This value matches the current address found by `cast call 0x84457ca9d0163fbc4bbfe4dfbb20ba46e48df254 "FINALIZATION_PERIOD_SECONDS()(uint256)" -r https://ethereum-sepolia.publicnode.com`

- **Key:** `0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc` <br/>
  **Before:** `0x000000000000000000000000afeac3ccabcbcb93e0d04fb0337b519360e898b8` <br/>
  **After:** `0x000000000000000000000000f243bed163251380e78068d317ae10f26042b292` <br/>
  **Meaning:** Implementation address is set to the new `L2OutputOracle`. The address can be found in the [Superchain Registry](https://github.com/ethereum-optimism/superchain-registry/blob/5ad42cbb49472a0bf164ade976426f7526ee6dfe/superchain/implementations/networks/sepolia.yaml#L8).

> ### `0x49Ff2C4be882298e8CA7DeCD195c207c42B45F66` (`OptimismMintableERC20FactoryProxy`)
> 
> Links:
> - [Etherscan](https://sepolia.etherscan.io/address/0x49Ff2C4be882298e8CA7DeCD195c207c42B45F66)

Reviewer: ```suggestion
### `0xb1efB9650aD6d0CC1ed3Ac4a0B7f1D5732696D37` (`OptimismMintableERC20FactoryProxy`)

Links:
- [Etherscan](https://sepolia.etherscan.io/address/0xb1efB9650aD6d0CC1ed3Ac4a0B7f1D5732696D37)
```

- [Superchain Registry](https://github.com/ethereum-optimism/superchain-registry/blob/5ad42cbb49472a0bf164ade976426f7526ee6dfe/superchain/extra/addresses/sepolia/metal.json#L7)

State Changes:
- **Key:** `0x0000000000000000000000000000000000000000000000000000000000000000` <br/>

>   **Before:** `0x00000000000000000000fd0bf71f60660e2f608ed56e1659c450eb1131200002` <br/>

Reviewer: Similar comment about prior value being an address, which is surprising and unlike the prior upgrades, so let's check this one out also

Author: Looks like spacers were added to `packages/contracts-bedrock/src/universal/OptimismMintableERC20Factory.sol` from the deployment to MCP [comparison](https://github.com/ethereum-optimism/optimism/compare/a7ff5a811612fa338d0a6d6dd72dc2ec9badef6d..e6ef3a900c42c8722e72c2e2314027f85d12ced5).

Author: Only additions, so it looks safe

- **Key:** `0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc` <br/>
  **Before:** `0x0000000000000000000000002f36789426d2e32d33d2fc2f2a7c06e5b1ed17f6` <br/>
  **After:** `0x000000000000000000000000e01efbeb1089d1d1db9c6c8b135c934c0734c846` <br/>
  **Meaning:** Implementation address is set to the new `OptimismMintableERC20Factory` implementation. The address can be found in the [Superchain Registry](https://github.com/ethereum-optimism/superchain-registry/blob/5ad42cbb49472a0bf164ade976426f7526ee6dfe/superchain/implementations/networks/sepolia.yaml#L10).

### `0x5D335Aa7d93102110879e3B54985c5F08146091E` (`L1CrossDomainMessengerProxy`)

todo: actual state changes 4 vs 3 in the template

Links:
- [Etherscan](https://sepolia.etherscan.io/address/0x5D335Aa7d93102110879e3B54985c5F08146091E)

> - [Superchain Registry](https://github.com/ethereum-optimism/superchain-registry/blob/5ad42cbb49472a0bf164ade976426f7526ee6dfe/superchain/extra/addresses/sepolia/metal.json#L3)

Reviewer: should Cmd+F for `metal` and repalce with `base` (and verify it still points to the right line), there's a few other "metal" references when I cmd+F

State Changes:

> todo: is this one set to true? value doesn't seem to indicate that

Reviewer: The linked storage layout for slot 0 is
```json
  {
    "bytes": "20",
    "label": "spacer_0_0_20",
    "offset": 0,
    "slot": "0",
    "type": "address"
  },
  {
    "bytes": "1",
    "label": "_initialized",
    "offset": 20,
    "slot": "0",
    "type": "uint8"
  },
  {
    "bytes": "1",
    "label": "_initializing",
    "offset": 21,
    "slot": "0",
    "type": "bool"
  },
  ```
  So the right most 20 bytes are a spacer, and the initialized byte comes after. The old version of this contract probably had `reinitializer(2)`, like the prior ones, which would explain the value there before. But let's confirm the prior value

Author: [Comparing](https://github.com/ethereum-optimism/optimism/compare/a7ff5a811612fa338d0a6d6dd72dc2ec9badef6d..e6ef3a900c42c8722e72c2e2314027f85d12ced5) the Base Sepolia deployment with the MCP contracts, it does show the `reinitializer(2)` is present in the initial deployment.

[batch-inbox-registry]: https://github.com/ethereum-optimism/superchain-registry/blob/5ad42cbb49472a0bf164ade976426f7526ee6dfe/superchain/configs/sepolia/metal.yaml#L9
[batch-inbox-etherscan]: https://sepolia.etherscan.io/address/0x24567B64a86A4c966655fba6502a93dFb701E316
[l1-standard-bridge-registry]: https://github.com/ethereum-optimism/superchain-registry/blob/a62fd38c9239459b27ea50a2df895e89a3305382/superchain/extra/addresses/sepolia/base.json#L5
[l1-standard-bridge-etherscan]: https://sepolia.etherscan.io/address/0xfd0Bf71F60660E2f608ed56e1659C450eB113120
[factory-registry]: https://github.com/ethereum-optimism/superchain-registry/blob/a62fd38c9239459b27ea50a2df895e89a3305382/superchain/extra/addresses/sepolia/base.json#L7
[factory-etherscan]: https://sepolia.etherscan.io/address/0xb1efB9650aD6d0CC1ed3Ac4a0B7f1D5732696D37
[output-oracle-registry]: https://github.com/ethereum-optimism/superchain-registry/blob/a62fd38c9239459b27ea50a2df895e89a3305382/superchain/extra/addresses/sepolia/base.json#L6
[output-oracle-etherscan]: https://sepolia.etherscan.io/address/0x84457ca9D0163FbC4bbfe4Dfbb20ba46e48DF254

> todo: this looks like an initialization but it wasn't present in the template

Reviewer: same `reinitializer` reason

Author: same thing in `SystemConfig.sol`: https://github.com/ethereum-optimism/optimism/compare/a7ff5a811612fa338d0a6d6dd72dc2ec9badef6d..e6ef3a900c42c8722e72c2e2314027f85d12ced5

- **Key:** `0x0000000000000000000000000000000000000000000000000000000000000000` <br/>
  **Before:** `0x0000000000000000000000000000000000000000000000000000000000000002` <br/>
  **After:** `0x0000000000000000000000000000000000000000000000000000000000000001` <br/>

>   **Meaning:** 

Reviewer: placeholder TODO for this + any other applicalble reinitializer changes that we need to explain

- **Key:** `0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc` <br/>
  **Before:** `0x0000000000000000000000001126e5afb588d39c3c5465a15af389146d309581` <br/>
  **After:** `0x000000000000000000000000ba2492e52f45651b60b8b38d4ea5e2390c64ffb1` <br/>
  **Meaning:** Implementation address is set to the new `SystemConfig` per the [Superchain Registry][system-config-registry] and [Etherscan][system-config-etherscan].

> todo: this one has 12 events vs 5 events in the template, but it should be fine

Reviewer: Interesting, what are the different events?

Author: These are it, seem like they were messing around with different values

https://sepolia.etherscan.io/address/0xf272670eb55e895584501d564AfEB048bEd26194#events

### `0xfd0Bf71F60660E2f608ed56e1659C450eB113120` (`L1StandardBridgeProxy`)

todo: simulated state changes 4 vs 5 in template

Links:
- [Etherscan](https://sepolia.etherscan.io/address/0xfd0Bf71F60660E2f608ed56e1659C450eB113120)
- [Superchain Registry](https://github.com/ethereum-optimism/superchain-registry/blob/a62fd38c9239459b27ea50a2df895e89a3305382/superchain/extra/addresses/sepolia/base.json#L5)

State Changes:
- **Key:** `0x0000000000000000000000000000000000000000000000000000000000000000` <br/>
  **Before:** `0x0000000000000000000000000000000000000000000000000000000000000003` <br/>
  **After:** `0x0000000000000000000000000000000000000000000000000000000000000001` <br/>

>   **Meaning:** The `initialized` boolean is set to `true`. The correctness of

Reviewer: Prior value of 3 is again likely because of the `reinitializer` change

Author: Now the L1StandardBridge was [upgraded](https://github.com/base-org/contract-deployments/blob/26783846e57316b02ed511d75edf4bf92dcc4269/sepolia/2023-10-07-upgrade-l1-standard-bridge/.env) after initial deployment. Here's the [comparison](https://github.com/ethereum-optimism/optimism/compare/e3ba24e72085d85bb5584dda33a03ccf60db86f0..e6ef3a900c42c8722e72c2e2314027f85d12ced5) from that commit and MCP.

Author: changes look safe

  **After:** `0x000000000000000000000000c2be75506d5724086deb7245bd260cc9753911be` <br/>
  **Meaning:** Sets `superchainConfig` at slot `0x32` (50). The correctness of
   this slot is attested to in the Optimism repo at [storageLayout/L1StandardBridge.json](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.3.0/packages/contracts-bedrock/snapshots/storageLayout/L1StandardBridge.json#L58-L64). The `superchainConfig` address can be found in the [Superchain Registry](https://github.com/ethereum-optimism/superchain-registry/blob/5ad42cbb49472a0bf164ade976426f7526ee6dfe/superchain/configs/sepolia/superchain.yaml#L8).

- **Key:** `0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc` <br/>
  **Before:** `0x000000000000000000000000acd2e50b877ab12924a766b8ddbd9272402c7d72` <br/>
  **After:** `0x00000000000000000000000064b5a5ed26dcb17370ff4d33a8d503f0fbd06cff` <br/>
  **Meaning:** Implementation address is set to the new `L1StandardBridge`. The address can be found in the [Superchain Registry](https://github.com/ethereum-optimism/superchain-registry/blob/5ad42cbb49472a0bf164ade976426f7526ee6dfe/superchain/implementations/networks/sepolia.yaml#L6).

> The only other state change is a nonce increment of `0x4672425c27a942bb27e7b9709c1b21ab89a3ca13`,
> which is the account sending the transaction and is the only signer on the safe.

Reviewer: since it's a 3/7 safe, we can be a bit more vague here
```suggestion
The only other state change is a nonce increment of the account being used to simulate the transaction.
```
